pallets/asset/src/lib.rs

```diff
@@ -990,14 +979,16 @@ decl_module! {
         pub fn update_identifiers(
             origin,
             ticker: Ticker,
-            identifiers: Vec<(IdentifierType, AssetIdentifier)>
+            identifiers: Vec<AssetIdentifier>
         ) -> DispatchResult {
             let sender = ensure_signed(origin)?;
             let did = Context::current_identity_or::<Identity<T>>(&sender)?;
             ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
-            for (typ, val) in &identifiers {
-                <Identifiers>::insert((ticker, typ.clone()), val.clone());
-            }
+            let identifiers: Vec<AssetIdentifier> = identifiers
+                .into_iter()
+                .filter_map(|identifier| identifier.validate())
+                .collect();
+            <Identifiers>::insert(ticker, identifiers.clone());
```

Comment:
Code above duplicated here; could be refactored into a function.

Reply:
I think it's fine it's only duped once.